apps/examples/smartfs/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,521 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ *
+ * apps/examples/smartfs/fs_performance/fs_performance_main.c
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <tinyara/timer.h>
+
+/****************************************************************************
+ * Pre Processor Definitions
+ ****************************************************************************/
+
+#define TEST_FILE			CONFIG_MOUNT_POINT"test_file"
+#define TEST_FLAG_FILE_CREATION		1
+#define TEST_FLAG_OVERWRITE		2
+#define TEST_FLAG_READ			3
+#define TEST_FLAG_GC_EFFECT		4
+#define TEST_FLAG_APPEND		5
+#define TEST_FILE_NAME_LEN_MAX		15
+#define ITR_OVE				5
+#define ITR_APP				5
+#define ITR_READ			100
+#define ITR_GC				10
+#define N_MULTIPLE_FILES		2
+#define TIMER_DEVNAME			"/dev/timer%d"
+
+/* Maximum buffer size used to read a file is 32kb */
+#define MAX_RD_BUF_SIZE			32768
+
+/* Maximum file size used for TCs is 64kb file */
+#define MAX_BUF_SIZE			65536
+#define MAX_FILE_SIZE			65536
+
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+
+static int g_fsz;
+static int g_bsz;
+static char g_file[N_MULTIPLE_FILES][TEST_FILE_NAME_LEN_MAX];
+static char g_rw_buf[MAX_BUF_SIZE];
+static char g_read_buf[MAX_RD_BUF_SIZE];
+static char g_timer_path[_POSIX_PATH_MAX];
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int do_test(int option)
+{
+	int ret = OK;
+	int i;
+	int j;
+	int fd;
+	unsigned long tot_time = 0;
+	unsigned long itr_time = 0;
+	struct timer_status_s before;
+	struct timer_status_s after;
+	int frt_fd = -1;
+
+	memset(g_rw_buf, '0', MAX_BUF_SIZE);
+	frt_fd = open(g_timer_path, O_RDONLY);
+	if (frt_fd < 0) {
+		printf("Unable to open file: %s errno: %d\n", g_timer_path, errno);
+		return ERROR;
+	}
+
+	ret = ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN);
```

Comment:
Please make a dependency with timer config in Kconfig.

Reply:
Updated